src/macaron/slsa_analyzer/analyzer.py

```diff
@@ -540,6 +541,14 @@ def to_analysis_target(config: Configuration, available_domains: list[str]) -> A
             except ValueError as error:
                 raise InvalidPURLError(f"Invalid input PURL: {config.get_value('purl')}") from error
 
+        # Validate the purl object, which is user-controllable.
+        if parsed_purl:
+            purl_str = parsed_purl.to_string()
```

Comment:
If we are only trying to sanitize the original purl_string provided, do you think we should do it before trying to create a PackageURL object from that purl string?